test/unit/eigensolver/mc/test_reduction_to_band.cpp

```diff
@@ -0,0 +1,320 @@
+    for (SizeType i = 0; i < from.size().rows(); ++i)
+      to({j, i}) = dlaf::conj(from({i, j}));
+}
+
+// band_size in elements
+template <class T>
+void setup_sym_band(MatrixLocal<T>& matrix, const SizeType& band_size) {
+  DLAF_ASSERT(band_size == matrix.blockSize().rows(), "not yet implemented", band_size,
+              matrix.blockSize().rows());
+  DLAF_ASSERT(band_size % matrix.blockSize().rows() == 0, "not yet implemented", band_size,
+              matrix.blockSize().rows());
+
+  DLAF_ASSERT(square_blocksize(matrix), matrix.blockSize());
+  DLAF_ASSERT(square_size(matrix), matrix.blockSize());
+
+  const auto k_diag = band_size / matrix.blockSize().rows();
+
+  // setup band "edge" and its tranposed
+  for (SizeType j = 0; j < matrix.nrTiles().cols(); ++j) {
+    const GlobalTileIndex ij{j + k_diag, j};
+
+    if (!ij.isIn(matrix.nrTiles()))
+      continue;
+
+    const auto& tile_lo = matrix.tile(ij);
+
+    // setup the strictly lower to zero (if there is any)
+    if (std::min(tile_lo.size().rows(), tile_lo.size().cols()) - 1 > 0) {
+      // clang-format off
+      lapack::laset(
+          lapack::MatrixType::Lower,
+          tile_lo.size().rows() - 1, tile_lo.size().cols() - 1,
+          0, 0,
+          tile_lo.ptr({1, 0}), tile_lo.ld());
+      // clang-format on
+    }
+
+    copy_transposed(matrix.tile_read(ij), matrix.tile(common::transposed(ij)));
+  }
+
+  // setup zeros in both lower and upper out-of-band
+  for (SizeType j = 0; j < matrix.nrTiles().cols(); ++j) {
+    for (SizeType i = j + k_diag + 1; i < matrix.nrTiles().rows(); ++i) {
+      const GlobalTileIndex ij{i, j};
+      if (!ij.isIn(matrix.nrTiles()))
+        continue;
+
+      dlaf::matrix::test::set(matrix.tile(ij), 0);
+      dlaf::matrix::test::set(matrix.tile(common::transposed(ij)), 0);
+    }
+  }
+
+  // mirror band (diagonal tiles are correctly set just in the lower part)
+  for (SizeType k = 0; k < matrix.nrTiles().rows(); ++k) {
+    const GlobalTileIndex kk{k, k};
+    const auto& tile = matrix.tile(kk);
+
+    mirror_on_diag(tile);
+  }
+}
+
+template <class T>
+auto all_gather_taus(const SizeType k, const SizeType chunk_size, const SizeType band_size,
+                     std::vector<hpx::shared_future<common::internal::vector<T>>> fut_local_taus,
+                     comm::CommunicatorGrid comm_grid) {
+  using namespace dlaf::comm::sync;
+
+  std::vector<T> taus;
+  taus.reserve(to_sizet(k));
+
+  hpx::wait_all(fut_local_taus);
+  auto local_taus = hpx::unwrap(fut_local_taus);
+
+  DLAF_ASSERT(band_size == chunk_size, band_size, chunk_size);
+
+  const auto n_chunks = std::ceil(float(k) / chunk_size);
+
+  for (auto index_chunk = 0; index_chunk < n_chunks; ++index_chunk) {
+    const auto owner = index_chunk % comm_grid.size().cols();
+    const bool is_owner = owner == comm_grid.rank().col();
+
+    const auto this_chunk_size = std::min(k - index_chunk * chunk_size, chunk_size);
+
+    std::vector<T> chunk_data;
```

Comment:
do we need to allocate memory with `reserve` here as well?

Reply:
if you look at what happens to `chunk_data`, you see that it is not needed.

In one branch, you have the copy from `local_taus`, which before copying elements should (afaik) resize the destination before performing the actual copy.

The else branch does the resize, which is slightly different than `reserve`, because it actually enlarges the container instead of just allocating memory underneath.